doc/source/development/window.rst

```diff
@@ -0,0 +1,187 @@
+Generalized Window Operations
+=============================
+
+Rationale
+---------
+The current window implementation in :file:`window.pyx` is difficult to
+customize and couples the creation of window bounds, loop over the values to
+aggregate, the logic for updating an aggregation.
```

Comment:
I'll mention this, but I do think it's a bit distracting from the argument since we will not be explicitly addressing that here though ideally in the future this infrastructure is reusable there, and in fact I'm designing this in such a way that that is the case.

Reply:
In any event, you're right that it's worth mentioning.

Reply:
Agreed that it is worth mentioning.

Reply:
It is maybe worth moving all groupby related discussion to a second doc/draft that builds on this one.

Reply:
(And leave that for possible future work).